rmf_fleet_adapter/src/rmf_fleet_adapter/agv/EasyFullControl.cpp

```diff
@@ -1400,6 +1461,101 @@ void EasyFullControl::EasyRobotUpdateHandle::update(
     });
 }
 
+//==============================================================================
+double EasyFullControl::EasyRobotUpdateHandle::max_merge_waypoint_distance()
+const
+{
+  auto updater = _pimpl->updater;
+  if (!updater->handle)
+  {
+    // If no handle is configured, return the default fleet value
+    return updater->nav_params->max_merge_waypoint_distance;
+  }
+
+  auto context =
+    RobotUpdateHandle::Implementation::get(*_pimpl->updater->handle).get_context();
+  return context->nav_params()->max_merge_waypoint_distance;
+}
+
+//==============================================================================
+void EasyFullControl::EasyRobotUpdateHandle::set_max_merge_waypoint_distance(
+  double distance)
+{
+  auto updater = _pimpl->updater;
+  if (!updater->handle)
+  {
+    return;
```

Comment:
It's not okay for these setters to silently have no effect. Users would be very confused if calling a function seems to randomly have no effect.

I'm going to refresh myself on what situations `updater->handle` might be a `nullptr` and see if we can tighten this up.

Reply:
Fixed by [91a5270](https://github.com/open-rmf/rmf_ros2/pull/289/commits/91a5270b4da219b20ad9b95cf93e024f0a21a857)